### PR TITLE
feat: record metrics on an interval in central store

### PR DESCRIPTION
**Pull request by VinozzZ**
## Which problem is this PR solving?

Record central store metrics on an interval so we have metrics ready for stress level calculation

## Short description of the changes

- added metrics for smart wrapper, central collectors
- added metrics cycle to run `RecordMetrics` in an interval
- add span counts to all spans' metadata instead of only on root span
- removed unnecessary code
- inject StressRelief to central collector

### CODE REVIEW

File: collect/central_collector.go [hunk 1]
```diff
@@ -101,24 +103,48 @@ func (c *CentralCollector) Start() error {
...
+	c.metricsCycle = NewCycle(c.Clock, c.Config.GetSendTickerValue(), c.done)
 	c.processorCycle = NewCycle(c.Clock, collectorCfg.GetProcessTracesPauseDuration(), c.done)
 	c.deciderCycle = NewCycle(c.Clock, collectorCfg.GetDeciderPauseDuration(), c.done)
 
 	c.Metrics.Register("collector_processor_batch_count", "histogram")
 	c.Metrics.Register("collector_decider_batch_count", "histogram")
 	c.Metrics.Register("trace_send_kept", "counter")
+	c.Metrics.Register("trace_duration_ms", "histogram")
+	c.Metrics.Register("trace_span_count", "histogram")
+	c.Metrics.Register("trace_decision_ket", "counter")
```
Comment (kentquirk) [inline collect/central_collector.go:115]: what does "ket" mean?

File: collect/central_collector.go [hunk 2]
```diff
@@ -630,6 +656,10 @@ func (c *CentralCollector) send(status *centralstore.CentralTraceStatus) {
 			sp.Data["meta.refinery.send_reason"] = TraceSendLateSpan
 
 		}
+		sp.Data["meta.span_event_count"] = int(status.SpanEventCount())
```
Comment (kentquirk) [inline collect/central_collector.go:659]: Did we decide to add counts to every span? And make it not optional? We'll have to change this in config too.

File: collect/central_collector_test.go [hunk 3]
```diff
@@ -681,11 +679,11 @@ func TestCentralCollector_AddCountsToRoot(t *testing.T) {
 
 	transmission.Mux.RLock()
 	require.Equal(t, 5, len(transmission.Events), "adding a root span should send all spans in the trace")
-	assert.Equal(t, nil, transmission.Events[0].Data["meta.span_count"], "child span metadata should NOT be populated with span count")
-	assert.Equal(t, nil, transmission.Events[1].Data["meta.span_count"], "child span metadata should NOT be populated with span count")
-	assert.Equal(t, nil, transmission.Events[1].Data["meta.span_event_count"], "child span metadata should NOT be populated with span event count")
-	assert.Equal(t, nil, transmission.Events[1].Data["meta.span_link_count"], "child span metadata should NOT be populated with span link count")
-	assert.Equal(t, nil, transmission.Events[1].Data["meta.event_count"], "child span metadata should NOT be populated with event count")
+	assert.Equal(t, 2, transmission.Events[0].Data["meta.span_count"], "child span metadata should NOT be populated with span count")
```
Comment (kentquirk) [inline collect/central_collector_test.go:682]: The error messages here and below are now wrong

File: collect/central_collector_test.go [hunk 4]
```diff
@@ -1322,16 +1320,15 @@ func duration(s string) config.Duration {
 
 func waitUntilReadyToDecide(t *testing.T, coll *CentralCollector, traceIDs []string) {
 	ctx := context.Background()
+	idMap := make(map[string]struct{})
```
Comment (kentquirk) [inline collect/central_collector_test.go:1323]: If you use generics.Set here you can call idset.Add(ids...).

**Review by kentquirk — APPROVED** [review]
A couple of nits, but feel free to merge after fixing the error messages.

File: collect/central_collector.go (hunk 2 above)
Comment (VinozzZ) [reply collect/central_collector.go:659]: I believe so unless my memory is messing with me which is not a uncommon thing

File: collect/central_collector.go (hunk 1 above)
Comment (VinozzZ) [reply collect/central_collector.go:115]: it means I had a typo 😆

**Pull request merged**